ral-facilities/datagateway
# Migrate lerna to yarn workspaces

## Description
Previously, we used lerna for monorepo support in the top level `package.json`, however it seems lerna is becoming unmaintained. This PR migrates all the commands previously implemented using lerna, to use yarn workspaces instead. 

The yarn plugin `workspace-tools` has been installed and used for the [`foreach`](https://yarnpkg.com/cli/workspaces/foreach) option, which runs the command for each package specified by the `workspaces` section in `package.json`. Note that this plugin is installed by default with yarn 4, so this may need to be removed when upgrading yarn ([see](https://www.npmjs.com/package/@yarnpkg/plugin-workspace-tools)).

The `pre-commit` script also used lerna, so it [has also been updated](https://github.com/ral-facilities/datagateway/pull/1434/files#diff-d2bc4bbf14eadc292d84e0ef5f7a93115c23557f533809fc80b896934291529d).

I have not removed `lerna-update-wizard` since it is a dev dependency and [works with yarn workspaces](https://www.npmjs.com/package/lerna-update-wizard).

Using yarn workspaces seems to generally be faster than with lerna, however execution times vary... 

## Testing instructions
Check the following commands inside top level `datagateway` package:

- [ ] `yarn run build`
- [ ] `yarn run lint`
- [ ] `yarn run tsc`
- [ ] `yarn run test:unit`
- [ ] `yarn run test:e2e`

Also check the `pre-commit` script is working.

## Agile board tracking
closes #1114
